src/05-derivatives/01-introduction.md

```diff
@@ -0,0 +1,296 @@
+If this file was generated with prior knowledge from additional sources
+(e.g., say the same subject's `T1w`), then the first item of `RawSources`
+should be the original `bold` file that defined the coordinate system:
+
+```JSON
+{
+    "RawSources": [
+        "sub-01/func/sub-01_task-rest_bold.nii.gz",
+        "sub-01/anat/sub-01_T1w.nii.gz"
+    ]
+}
+```
+
+## Metadata conventions
+
+-   Unless specified otherwise, individual sidecar JSON files and all metadata
+    fields within are OPTIONAL. However, the appropriate use of these files and
+    pertinent fields is very valuable and thus encouraged. Moreover, for some
+    types of files, there may be one or more required metadata fields, in which
+    case at least one metadata file containing that field must be located
+    somewhere within the file’s hierarchy (per [the Inheritance
+    Principle](../02-common-principles.md#the-inheritance-principle)).
+
+-   When chaining derivative pipelines, any JSON fields that were specified as
+    mandatory in the input files SHOULD be propagated forward in the output
+    file’s JSON provided they remain valid. Non-required JSON fields MAY be
+    propagated, and are highly useful, but it is the pipeline’s responsibility
+    to ensure that the values are still relevant and appropriate to the type of
+    output data.
+
+## File naming conventions
+
+-   Filenames that are permissible for a raw BIDS data type have a privileged
+    status. Any modification of raw files must use a modified filename that does
+    not conflict with the raw filename. Further, any files created as part of a
+    derivative dataset must not match a permissible filename of a valid raw
+    dataset. Stated equivalently, if any filename in a derivative dataset has a
+    name permissible for a raw BIDS data, then that file must be an identical
+    copy of that raw file.
+
+-   Each Derivatives filename MUST be of the form:
+    `<source_keywords>[_keyword-<value>]_<suffix>.<ext>`
+    (where `<value>` could either be an `<index>` or a `<label>` depending on
+    the keyword)
+
+-   When the derivatives chain involves outputs derived from a single raw input,
+    `source_keywords` MUST be the entire source filename, with the omission of
+    the source suffix and extension. One exception to this rule is filename
+    keywords that are no longer relevant. Depending on the nature of the
+    derivative file, the suffix can either be the same as the source file if
+    that suffix is still appropriate, or a new appropriate value selected from
+    the controlled list.
+
+-   There is no prohibition against identical filenames in different derived
+    datasets, although users should be aware of the potential ambiguity this can
+    create and use the sidecar JSON files to detail the specifics of individual
+    files.
+
+-   When necessary to distinguish two files, the `_desc-<label>` keyword-value
+    should be used. This includes the cases of needing to distinguish both
```

Comment:
I'm not sure that this should be a SHOULD in the very general case of "needing to distinguish two files". For instance, say a pipeline generates some derivative data on two different voxel grids, the T1w and the fMRI. So you might have "`_space-T1w`" and "`_space-orig`" in your two otherwise identical file names. This current wording seems to suggest that you SHOULD still nevertheless shoehorn a "`_desc-*`" into each filename. Maybe a more sensible suggestion would be that if there is no more appropriate dictionary keyword by which to disambiguate between two files, "`_desc-*`" should then be utilised?

Reply:
Changing to:

> "When necessary to distinguish two files that do not otherwise have a distinguishing entity, the `_desc-<label>` keyword-value SHOULD be used."